libdisco/asymmetric_test.go

```diff
@@ -7,21 +7,68 @@ import (
 
 func TestSignVerify(t *testing.T) {
 	input := []byte("hi, how are you?")
-	kp := GenerateSigningKeyPair()
+	kp, err := GenerateSigningKeypair()
+
+	if err != nil {
+		t.Fatal("failed to generate a signing keypair")
+	}
 	sig := kp.Sign(input)
 
-	if kp.Verify(input, sig) != nil {
-		t.Fatal("Schnorr doesn't work")
+	if ok, err := kp.Verify(input, sig); !ok || err != nil {
```

Comment:
not a fan of ok + err, it makes the caller check two variables instead of one. I'd prefer the original API which is harder to misuse.

Reply:
The verification can fail without an actual error, the signature can be wrong.
The trade-off here is by adding an error type each for bad signatures, malformed signatures 
or letting the user manage the error handling which is preferred.

Reply:
I'm not sure what is the benefit of letting the user figure out how to react to different types of errors. A signature is either good or bad, if it's bad it should not matter why it's bad.